internal/proto/local/plugin/host.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+
+package plugin.sdk.proto;
+
+option go_package = "github.com/hashicorp/boundary/internal/plugin/sdk/proto;proto";
+
+import "google/protobuf/struct.proto";
+
+// Host contains all fields related to a host returned from a plugin.
+message Host {
+  // The primary address for the host result.
+  string address = 10;
+
+	// Name of the host.
+	string name = 20;
+
+	// The host catalog of which this host is a part.
+	string host_catalog_id = 30;
+
+	// The list of host sets containing this host.
+	repeated string host_set_ids = 40;
```

Comment:
The risk here is that the plugin will only be able to populate this from the set of host sets provided in the ListHosts request.  This will not necessarily be a complete list.  MVP we are not planning on populating the host_set list on the API resource, so I don't think this is needed yet.

Reply:
👍 Fair point, will remove for now. I think that we might need to figure out the best way to determine exactly what host is a member of what set for caching, maybe possibly by way of a mechanism that allows us to track what sets and catalogs a host has been "seen" in. I think that's the best way I can think of to actually make those kinds of associations that would be pretty hard-set in static hosts.

Given this logic and pursuant to the next comment It think we can just remove the ID from the payload too.